scripts/unseal.py

```diff
@@ -0,0 +1,29 @@
+import json
+import os
+import requests
+import sys
+sys.path.append('scripts/')
+from pyinfra import host, local
+from pyinfra.operations import python, server
+from status import status
+
+SHELL = '/usr/bin/env bash'
+
+# curl --request PUT --data '{"key": "abcd1234..."}' https://vault.$KITT_DOMAIN/v1/sys/unseal
+def unseal():
+    print('Unseal Moria')
+    url = "vault.{}".format(os.getenv("KITT_DOMAIN"))
+    for index in range(1, 4):
+        key = local.shell(f'pass moria/keys_{index} | base64 -d')
```

Comment:
i'm on linux and get this error when running `make unseal`

```
$ make unseal
--> Loading config...
--> Loading inventory...

--> Connecting to hosts...
    [@local] Connected

--> Preparing operations...
    Loading: scripts/unseal.py
Unseal Moria
Traceback (most recent call last):
  File "src/gevent/greenlet.py", line 854, in gevent._gevent_cgreenlet.Greenlet.run
  File "/home/mel/.local/lib/python3.8/site-packages/pyinfra/api/util.py", line 402, in read_buffer
    line = line.decode('utf-8')
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xaf in position 1: invalid start byte
2020-07-15T14:56:19Z <Greenlet at 0x7f9a72581ae0: read_buffer('stdout', <FileObjectPosix at 0x7f9a73f89400 _fobj=<_io.Buff, <Queue at 0x7f9a724a8d00 queue=deque([('stderr', ', print_output=False, print_func=<function read_buffers_into_queue.<locals>.<lambda)> failed with UnicodeDecodeError

--> [@local] pyinfra error: Local command failed: pass moria/keys_1 | base64 -d
['base64: invalid input']
make: *** [Makefile:70: unseal] Error 1

```

i'm pretty sure the pass moria/key_x vars aren't base64 encoded.

and on line 19 below, the data for that put request should look something like {"key": "1234" }

Reply:
this diff works for me:

```
diff --git a/scripts/unseal.py b/scripts/unseal.py
index 14ff70f..21fb6e8 100644
--- a/scripts/unseal.py
+++ b/scripts/unseal.py
@@ -14,9 +14,9 @@ def unseal():
     print('Unseal Moria')
     url = "vault.{}".format(os.getenv("KITT_DOMAIN"))
     for index in range(1, 4):
-        key = local.shell(f'pass moria/keys_{index} | base64 -d')
+        key = local.shell(f'pass moria/keys_{index}')
         try:
-            req = requests.put(f'https://{url}/v1/sys/unseal', data = f'{key}')
+            req = requests.put(f'https://{url}/v1/sys/unseal', data = '{{"key": "{}"}}'.format(key))
             req.raise_for_status()
         except requests.exceptions.RequestException as e:
             raise SystemExit(e)
```

Reply:
@bellaweo can you change it back to what works for you? I'll figure out why I had to make those changes.  What python are you running, 2 or 3?

Reply:
Python 3.8.2